### PR TITLE
CB-5658 CB-5659 fix: pagination resolving

### PR DESCRIPTION
**Pull request by Wroud**

### CODE REVIEW

File: webapp/packages/core-resource/src/Resource/ResourceAlias.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ export abstract class ResourceAlias<TKey, TOptions extends ResourceAliasOptions>
     return undefined;
   }
 
+  copy(): this {
```
Comment (devnaumov) [inline webapp/packages/core-resource/src/Resource/ResourceAlias.ts]: I think we should add some comments here to explain why we need this method

File: webapp/packages/core-navigation-tree/src/NodesManager/NavTreeResource.ts [hunk 2]
```diff
@@ -461,8 +461,8 @@ export class NavTreeResource extends CachedMapResource<string, string[], Record<
   }
 
   protected async loader(originalKey: ResourceKey<string>): Promise<Map<string, string[]>> {
-    const pageKey =
-      this.aliases.isAlias(originalKey, CachedResourceOffsetPageKey) || this.aliases.isAlias(originalKey, CachedResourceOffsetPageListKey);
+    const pageListKey = this.aliases.isAlias(originalKey, CachedResourceOffsetPageListKey);
+    const pageKey = this.aliases.isAlias(originalKey, CachedResourceOffsetPageKey) || pageListKey;
```
Comment (sergeyteleshev) [inline webapp/packages/core-navigation-tree/src/NodesManager/NavTreeResource.ts]: I can see a copy-paste here and a little bit below for all resources with pagination.
Maybe we should add a helper here so it returns the correct `pageKey` and the `page`? So we don't miss this bug when we create a new feature with the exact pagination mechanism

File: webapp/packages/core-resource/src/Resource/CachedResource.ts [hunk 3]
```diff
@@ -91,41 +91,58 @@ export abstract class CachedResource<
...
 
-        for (const page of pageInfo.pages) {
-          if (page.isHasCommonSegment(from, to)) {
-            keys.push(...page.get(from, to));
+          for (const page of pageInfo.pages) {
+            if (page.isHasCommonSegment(from, to)) {
+              for (const pageKey of page.get(from, to)) {
+                keys.add(pageKey);
+              }
+            }
```
Comment (sergeyteleshev) [inline webapp/packages/core-resource/src/Resource/CachedResource.ts:140]: Same here. we can define helper like `getKeysFromPages(pages)`

**Review by sergeyteleshev — APPROVED** [review]

File: webapp/packages/core-resource/src/Resource/ResourceAlias.ts (hunk 1 above)
Comment (Wroud) [reply webapp/packages/core-resource/src/Resource/ResourceAlias.ts]: removed

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/core-resource/src/Resource/CachedResource.ts (hunk 3 above)
Comment (Wroud) [reply webapp/packages/core-resource/src/Resource/CachedResource.ts:140]: let's keep it for now

File: webapp/packages/core-navigation-tree/src/NodesManager/NavTreeResource.ts (hunk 2 above)
Comment (Wroud) [reply webapp/packages/core-navigation-tree/src/NodesManager/NavTreeResource.ts]: now you can use `getOffsetPageKeyInfo` helper

**Review by sergeyteleshev — APPROVED** [review]

**Pull request merged**